docs/oras-auth-provider.md

```diff
@@ -0,0 +1,182 @@
+# ORAS Authentication Provider
+
+ORAS handles all referrer operations using registry as the referrer store. It uses authentication credentials to authenticate to a registry and access referrer artifacts. Ratify contains many Authentication Providers to support different authentication scenarios. The user specifies which authentication provider to use in the configuration.
+
+The `auth-provider` section of configuration file specifies the authentication provider. The `name` field is REQUIRED for Ratify to bind to the correct provider implementation. 
+
+## Example config.json
+```
+{
+    "stores": {
+        "version": "1.0.0",
+        "plugins": [
+            {
+                "name": "oras",
+                "localCachePath": "./local_oras_cache",
+                "auth-provider": {
+                    "name": "<auth provider name>",
+                    <other provider specific fields>
+                }
+            }
+        ]
+    },
+    "verifiers": {
+        "version": "1.0.0",
+        "plugins": [
+            {
+                "name":"notaryv2",
+                "artifactTypes" : "application/vnd.cncf.notary.v2.signature",
+                "verificationCerts": [
+                    "<cert folder>"
+                  ]
+            }  
+        ]
+    }
+}
+```
+
+
+NOTE: ORAS will attempt to use anonymous access if the authentication provider fails to resolve credentials.
+
+## Supported Providers
+
+### 1. Docker Config
+This is the default authentication provider. Ratify attempts to look for credentials at the default docker configuration path ($HOME/.docker/config.json) if the `auth-provider` section is not specified.
+
+Specify the `configPath` field for the `docker-config` authentication provider to use a different docker config file path. 
+
+```
+"stores": {
+        "version": "1.0.0",
+        "plugins": [
+            {
+                "name": "oras",
+                "localCachePath": "./local_oras_cache",
+                "auth-provider": {
+                    "name": "docker-config",
+                    "configPath": <custom file path string>
+                }
+            }
+        ]
+    }
+```
+
+### 2. Azure Workload Identity
+Ratify pulls artifacts from a private Azure Container Registry using Workload Federated Identity in an Azure Kubernetes Service cluster. For an overview on how workload identity operates in Azure, refer to the [documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/workload-identity-federation). 
+
```

Comment:
I ran into a below scenario, I suspect this is related to caching of the registry client. maybe we should add to store.authCache cache only after success?

1. Configured ratify pod with workload identity
2. Tried it verify an image , verification failed with 401 due to missing acr permissions on workload identity
3. Assigned workload identity with acrpull
4. tries to deploy an image, verification continues to fail
5. Ratify pod restarted, verification succeeded.

Expected to succeed at step 4 without restart.

Reply:
@susanshi thanks for pointing this out. The issue is in fact related to the auth cache. I'm proposing that we parse the various ORAS operation errors and if we see it's 401, we can evict the registry client from the auth cache. This will force a refetch of the credentials the next attempt.

Reply:
I will be merging this eviction change after ORAS go v2 migration is complete since it depends on the changes.